plenum/common/messages/fields.py

```diff
@@ -181,6 +188,14 @@ def _specific_validation(self, val):
                 return val_error
 
 
+class HetroMapField(FieldBase):
```

Comment:
I don't have strong opinion on this but Hetro makes it clear that types can be distinct, implying support for multiple types, `AnyMapField` indicates any type can be used but does not indicate heterogeneity.